### DISCUSSION TITLE
FEVM Deal Making

### DISCUSSION THREAD

**Discussion by wildanvin** [filecoin-project/community#625] [Developers]
Hello,

I am trying to make a deal with a Storage Provider in the Hyperspace network through the fevm. I am following this [starter-kit](https://github.com/filecoin-project/fvm-starter-kit-deal-making). I was able to compile the contract and deploy it to hyperspace, though I had to use this [Pull Request](https://github.com/filecoin-project/fvm-starter-kit-deal-making/pull/7) to make it work. 

I am using [Ligthhouse](https://data.lighthouse.storage/dashboard/view-file/d058cfe2-7ad2-4ad4-92c6-59d6024df52b) to get the required info that the contract needs to make a deal (link to car file, commP, piece size, car size).  

I am able to propose the deal but I think it never gets accepted by a SP. I am pretty sure of it because when I query the [`pieceStatus`](https://github.com/filecoin-project/fvm-starter-kit-deal-making/blob/main/client-contract/src/DealClient.sol#L100) mapping of the `DealClient` contract it always gives me `RequestSubmitted` but never `DealPublished` or `DealActivated`. Also the front end tells me that the deal ID is 0 which is obviously wrong (right?).

So I don´t know how to make the SP accept my deal. Should I add some FIL so it will pick it up? Is the SP Boost not working? 

Also is this the best way to make a deal in the FEVM?

This is my slightly modified version of the [repo](https://github.com/wildanvin/hypertally-fevm).

Any help or guidance would be much appreciated.
Thanks.

**Comment by CubikRuubik** [comment]
Hi @wildanvin , did you solve your issue with 0 ID? Have the same one(

**Reply by wildanvin** [reply]
hello, I haven't try recently to be honest but I didn't solve it one year ago :(

**Reply by CubikRuubik** [reply]
Oh, sadly. But thanks for reply!
